### PR TITLE
More integration catalogue improvements

### PR DESCRIPTION
**Pull request by delucis**
- Renamed “Loaders” category label to “Content Loaders”
- Added a new “Images + Media” category (accompanying docs PR: https://github.com/withastro/docs/pull/9598)
- Add a bunch of JSDoc types to stuff to help keep track of what’s happening where
- Validate npm registry API responses
- Updates the integration data in the repo with these changes

### CODE REVIEW

File: src/content/integrations/@astro-communityastro-embed-youtube.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: "@astro-community/astro-embed-youtube"
 title: "@astro-community/astro-embed-youtube"
 description: Component to easily embed YouTube videos on your Astro site
 categories:
-  - uncategorized
+  - css+ui
```
Comment (sarah11918) [inline src/content/integrations/@astro-communityastro-embed-youtube.md:6]: just checking you don't want this to be `media`? I mean, I guess you would know. 😄

Comment (delucis) [reply src/content/integrations/@astro-communityastro-embed-youtube.md:6]: This predates the media category. I’ll need to go add more keywords after this merges 😁

**Review by sarah11918 — APPROVED** [review]
These looks great to me! Absolutely BTWWHB.

CSS + UI is still a huge category, and maybe that's just because there are more of them. But scanning that category I see things like our official renderers showing up (and first): 
image

I'm not sure I think "CSS and UI" and I think of the frameworks? Even though yes, they are UI Frameworks according to docs. I don't know. Just feels weird.

But that can be a refinement for another day!

**Comment by delucis** [comment]
> I'm not sure I think "CSS and UI" and I think of the frameworks? Even though yes, they are UI Frameworks according to docs. I don't know. Just feels weird.

Agreed. Not sure what the decision was, but currently the `renderer` keyword adds things to both “Framework” and “CSS + UI”. It does kind of make sense though — after all React is a thing you might use to build your UI. Something we can definitely iterate on — easier to decide these categories now we _have_ all these integrations instead of just having to try to imagine what categories might be needed one day.

**Pull request merged**